doc/adr/0002-GraphQL.md

```diff
@@ -52,10 +52,20 @@ Expected [TSP][tsp] features support.
 
 #### Executing actions
 
-* Applying filter.
-* Launching analysis, e.g. follow CPU or Thread for Critical Path analysis.
+##### Applying filters
```

Comment:
> Filtering support in Trace Compass (Eclipse version) is currently provided in different ways:
> 
> * `Events Table` filtering (and searching)  is support by either a regular expression that can be entered per column or by supplying a ITmfFilter instance. There is also some implementation, where a regular expression can be converted to a regular expression and vice versa. This is used in conjunction with incubator global filtering.

Thanks for this. Should have already been covered (summarized) by the base commit 5824dfa. But please do let me know if not enough, and on which ADR lines exactly; so I know what to potentially amend.

> * Data Providers like the time graph data providers, have APIs to provide regular expressions, which the is applied when doing fetchState for example. This will tag the states for highlighting or removing on the UI level.

I didn't mention this other case yet in the ADR, indeed; will check this out. Thanks.

> In the Theia trace viewer right now, there is only column based searching implemented in the events table. Filtering on the view level is not implemented nor on global level (global level means, apply filter to all the views). The search filter is supplied as a regular expression to the fetchLines() call towards the events table data provider.

Ack, and thanks for the detailed explanation(s), always useful. Commit 5824dfa should cover this angle.

> To enhance searching and to add filtering support, the API calls on the TSP has to be designed. We could settle that for any filter, we use regular expressions which then will be added as query parameters to the relevant fetch() TSP calls. On the server side, this expression has to be further passed on the data provider implementation and proper handling is applied when returning the data.
> 
> A second approach which can additional to the above, we can define a filter endpoint to create filters on the server and which will have a ID, and clients can supply the filter ID to the fetch() query message.
> 
> GraphQL or REST TSP specification will have to provide support for supplying these filters to query calls. How it is in implemented on the back-end is then trace server specific.

Ack; thanks. Now this again should be kept for another (later) ADR scope I think.

Reply:
I didn't review https://github.com/theia-ide/theia-trace-extension/commit/5824dfa0d235b8fa6ff77f052416790d75ecda8b, so I only was able to read now what was written. I think that commit, has some basic information about filtering.

Reply:
> I didn't review [5824dfa](https://github.com/theia-ide/theia-trace-extension/commit/5824dfa0d235b8fa6ff77f052416790d75ecda8b), so I only was able to read now what was written. This commit has some basic information about filtering. We could add to that some information the 2 ways that I described:
> 
> * per view filtering (providing a filter expression to a specific data provider)
> * global filtering using filter endpoint

Will do, yes; relates to our previous thread further above. -Thanks again.

Regarding that past review, you were a co-reviewer and PR ended up being approved by our colleague. It was a simpler and more trivial one back then. We can always amend though (or iterate), which is what I'll do soon, per above.

> Of course, that might change once we tackle the filtering topic, but that should not change in GraphQL context.

Yes. Filtering will be skipped for now, as previously discussed, as it became out of this current ADR version scope.

Reply:
> > Of course, that might change once we tackle the filtering topic, but that should not change in GraphQL context.
> 
> Yes. Filtering will be skipped for now, as previously discussed, as it became out of this current ADR version scope.

Gah. What I meant was, the non-server parts of filtering will be skipped -for now.

Reply:
> > I didn't review [5824dfa](https://github.com/theia-ide/theia-trace-extension/commit/5824dfa0d235b8fa6ff77f052416790d75ecda8b), so I only was able to read now what was written. This commit has some basic information about filtering. We could add to that some information the 2 ways that I described:
> > 
> > * per view filtering (providing a filter expression to a specific data provider)
> > * global filtering using filter endpoint
> 
> Will do, yes; relates to our previous thread further above. -Thanks again.

I ended up not doing this for reasons explained in latest commit (message) pushed herein.

Reply:
> Gah. What I meant was, the non-server parts of filtering will be skipped -for now.

-Done.